### DISCUSSION TITLE
Monitor PHP-FPM via socket permission problem

### DISCUSSION THREAD

**Discussion by bessone** [netdata/netdata#12312] [Q&A] [answered]
Hello,

I have a problem with PHP-FPM monitoring via socket.

The netdata agent installation is the default, while debugging I get these messages:
```
[ DEBUG ] build[manager] build.go:153 received config group ('/etc/netdata/go.d/phpfpm.conf'): 1 jobs (added: 1, removed: 0)
[ DEBUG ] build[manager] build.go:302 building phpfpm[edom] job, config: map[__provider__:file reader __source__:/etc/netdata/go.d/phpfpm.conf autodetection_retry:0 module:phpfpm name:local priority:70000 socket:/run/php/mysocket.sock update_every:1]
[ DEBUG ] phpfpm[edom] init.go:35 using socket client: /run/php/mysocket.sock
[ DEBUG ] phpfpm[edom] init.go:36 using timeout: 1s
[ ERROR ] phpfpm[edom] phpfpm.go:66 error on connecting to socket '/run/php/mysocket.sock': dial unix /run/php/php7.4-fpm-edom.barbero.domains.sock: connect: permission denied
[ ERROR ] phpfpm[edom] job.go:157 check failed
```

The owner of the socket is the website user, for each site I have a different php-fpm pool.
The socket group is www-data, the netdata agent cannot access the file for this reason? Can it be done somehow?

Thanks!

**Comment by ilyam8** [comment]
Hi, @bessone. I [asked](https://github.com/netdata/go.d.plugin/issues/590#issuecomment-1058299965) a few Netdata users to help us with your issue. I know they used the php-fpm collector with a socket connection.

**Reply by glesys-andreas** [reply]
Hi, Sorry for the late reply.
Make sure that the netdata user can read the socket, either with correct permissions or add them to the www-data group.
Please let me know if you any more issues with the plugin, in that case i will setup a testserver and try it out.

**Reply by glesys-andreas** [reply]
Either one of these should work,:
Add netdata to www-data group (this is what i used):
`usermod -a -G  www-data netdata` 
Or, in you PHP FPM pool config add:
`listen.mode = 0666`

**Reply by ilyam8** [reply]
@glesys-andreas, thanks! Can you add this info as a separate message so I can mark it as an answer? Can't do it when it is a reply in a thread. I will update the collector readme.

**Comment by glesys-andreas** [comment, marked as answer]
@ilyam8  Sure!
Make sure that the netdata user can read the socket, either with correct permissions or add them to the www-data group.

Either one of these should work,:
Add netdata to www-data group (this is what i used):
`usermod -a -G www-data netdata`
Or, in you PHP FPM pool config add:
`listen.mode = 0666`
